nav2_behavior_tree/plugins/action/compute_path_through_poses_action.cpp

```diff
@@ -60,9 +62,19 @@ BT::NodeStatus ComputePathThroughPosesAction::on_cancelled()
   setOutput("path", empty_path);
   // Set empty error code, action was cancelled
   setOutput("error_code_id", ActionResult::NONE);
+  setOutput("error_msg", "");
   return BT::NodeStatus::SUCCESS;
 }
 
+void ComputePathThroughPosesAction::halt()
+{
+  nav_msgs::msg::Path empty_path;
+  setOutput("path", empty_path);
+  // DO NOT reset "error_code_id" output port, we want to read it later
```

Comment:
Please explain for here and `ComputePathToPosesAction` below. We have the error code ID properly populated before, yes? Then we shouldn't need to override the function at all

Reply:
Looking at the current main branch.
Why does ComputePathToPoseAction have a halt function that clears the path from the blackboard?  
Why does ComputePathThroughPosesAction not have a halt() function override?

I think these two actions are very close in implementation and was seeking to keep them in sync.  So I added the halt function to ComputePathThroughPosesAction for symmetry.

Prior to this feature branch ComputePathToPoseAction was the only action that actually implements/overrides the halt() function.

While creating these feature I was over zealous at following the pattern of calling setOutput() to clear the path, in the halt() function to also clear the error_code_id and error_msg.

I have been struggling, during creation of this branch, with the error_code_id and error_msg being cleared, for some actions (Couldn't put my finger on which ones at the time), before they are able to be read by the navigators and propogated back to the navigator clients.  There has been much hair pulling, tmux split window debugging and ALOT of RCLCPP_WARNs throughout the various async callback chains to finally come up with these two comments to remind the future NOT clear the error_code_id and error_msg in the halt() function.

Reply:
> Why does ComputePathToPoseAction have a halt() function that clears the path from the blackboard? Why does ComputePathThroughPosesAction not have a halt() function override?

Ah, that looks good :+1: